app/views/common/os_selection/_pxe_loader.html.erb

```diff
@@ -0,0 +1,12 @@
+<%= fields_for item do |f| -%>
+  <% loaders = @operatingsystem.available_loaders(item.architecture); if loaders.any? -%>
+    <% selected_attr = f.object.new_record? && item.hostgroup.present? && item.hostgroup.pxe_loader.nil? ? { :selected => @operatingsystem.preferred_loader(item.architecture) } : {} -%>
```

Comment:
I think I have a solution - instead of storing the filename itself, I will simply store the loader name. Lot of complex handling will become easier. It will be only translated to filename when needed by TFTP orchestration.